RELEASING.md

```diff
@@ -40,3 +40,33 @@ You should have no local changes and tests should be passing.
 1. Close the milestone for this release. https://github.com/tigera/operator/milestones
 
 1. Go to https://github.com/tigera/operator/milestones and create any new milestones that should exist (e.g., next patch release)
+
+## Creating a release on RH Catalog
```

Comment:
Should we include something that only releases targeting Calico (not Enterprise) should be added to the RH Catalog?

Reply:
556c0655aca9c858cdac91e71b563de44a1ff432